src/io/s3_filesys.cc

```diff
@@ -824,10 +824,10 @@ void ListObjects(const URI &path,
 S3FileSystem::S3FileSystem() {
   const char *keyid = getenv("AWS_ACCESS_KEY_ID");
   const char *seckey = getenv("AWS_SECRET_ACCESS_KEY");
-  if (keyid == NULL) {
+  if (keyid == NULL || !*keyid) {
     LOG(FATAL) << "Need to set enviroment variable AWS_ACCESS_KEY_ID to use S3";
   }
-  if (seckey == NULL) {
+  if (seckey == NULL || !*seckey) {
```

Comment:
My implementation returns NULL (Ubuntu). However, the documentation is not specific on this and states that it returns NULL if the environment variable is not found.  However, I can put an empty variable in the environment:
[coolivie@DEV:~]export FOO=
[coolivie@DEV:~]env | grep FOO
FOO=
[coolivie@DEV:~]